docs/content/languages/python.md

```diff
@@ -0,0 +1,356 @@
+---
+title: "Using Python with Fission"
+weight: 10
+---
+
+Fission supports functions written in Python.  Both Python 2.x and
+Python 3.x are supported.  In this usage guide we'll cover how to set
+up and use a Python environment on Fission, write functions, and work
+with dependencies.  We'll also cover basic troubleshooting.
+
+## Before you start
+
+We'll assume you have Fission and Kubernetes setup.  If not, head over
+to the [installation guide](/latest/installation).  Verify your Fission setup with:
+
+```
+$ fission --version
+```
+
+## Add the Python environment to your cluster
+
+Fission language support is enabled by creating an _Environment_.  An
+environment is the language-specific part of Fission.  It has a
+container image in which your function will run.
+
+```
+$ fission environment create --name python --image fission/python-env 
+```
+
+## Create a simple function in Python
+
+Create a file named `hello.py`:
+
+```python
+def main():
+    return "Hello, world!"
+```
+
+Create a Fission function (this uploads the file to Fission on the
+cluster):
+
+```bash
+$ fission function create --name hello --env python --code hello.py 
+```
+
+Invoke this function through the Fission CLI:
+
+```bash
+$ fission function test --name hello
+Hello, world!
+```
+
+You can also invoke this function by creating an HTTP trigger and
+making an HTTP request to the Fission router.  Ensure you have your
+router's address in the `FISSION_ROUTER` environment variable as 
+[this guide describes](https://docs.fission.io/0.11.0/installation/env_vars/#fission-router-address).
+Then,
+
+```bash
+$ fission route create --method GET --url /hello --function hello 
+
+$ curl $FISSION_ROUTER/hello
+Hello, world!
+```
+
+## Function input and output interface
+
+In this section we'll describe the input and output interfaces of
+Python functions in Fission.  Fission's Python integration is built on
+the Flask framework.  You can access HTTP requests and responses as
+you do in Flask.  We'll provide some examples below.
+
+### Accessing HTTP Requests
+
+#### HTTP Headers
+
+Write a simple `headers.py` with something like this:
+
+```python
+from flask import request
+
+def main():
+    myHeader = request.headers['x-my-header']
```

Comment:
Maybe add error handling here otherwise user only gets 500 error if curl with proper header.
Following are logs from function pod.

```
[2018-12-03 06:56:21.88851898 +0000 UTC] 2018-12-03 06:56:21,886 - ERROR - Exception on / [GET]
[2018-12-03 06:56:21.888539108 +0000 UTC] Traceback (most recent call last):
[2018-12-03 06:56:21.888544375 +0000 UTC]   File "/usr/lib/python3.5/site-packages/flask/app.py", line 1982, in wsgi_app
[2018-12-03 06:56:21.88854959 +0000 UTC]     response = self.full_dispatch_request()
[2018-12-03 06:56:21.888681221 +0000 UTC]   File "/usr/lib/python3.5/site-packages/flask/app.py", line 1614, in full_dispatch_request
[2018-12-03 06:56:21.888691176 +0000 UTC]     rv = self.handle_user_exception(e)
[2018-12-03 06:56:21.888695767 +0000 UTC]   File "/usr/lib/python3.5/site-packages/flask/app.py", line 1517, in handle_user_exception
[2018-12-03 06:56:21.888700128 +0000 UTC]     reraise(exc_type, exc_value, tb)
[2018-12-03 06:56:21.8887043 +0000 UTC]   File "/usr/lib/python3.5/site-packages/flask/_compat.py", line 33, in reraise
[2018-12-03 06:56:21.88870849 +0000 UTC]     raise value
[2018-12-03 06:56:21.888747218 +0000 UTC]   File "/usr/lib/python3.5/site-packages/flask/app.py", line 1612, in full_dispatch_request
[2018-12-03 06:56:21.888753509 +0000 UTC]     rv = self.dispatch_request()
[2018-12-03 06:56:21.888758008 +0000 UTC]   File "/usr/lib/python3.5/site-packages/flask/app.py", line 1598, in dispatch_request
[2018-12-03 06:56:21.888762396 +0000 UTC]     return self.view_functions[rule.endpoint](**req.view_args)
[2018-12-03 06:56:21.888767904 +0000 UTC]   File "server.py", line 97, in f
[2018-12-03 06:56:21.888772288 +0000 UTC]     return self.userfunc()
[2018-12-03 06:56:21.888776592 +0000 UTC]   File "/userfunc/user", line 4, in main
[2018-12-03 06:56:21.888783982 +0000 UTC]     myHeader = request.headers['x-my-header']
[2018-12-03 06:56:21.888788254 +0000 UTC]   File "/usr/lib/python3.5/site-packages/werkzeug/datastructures.py", line 1354, in __getitem__
[2018-12-03 06:56:21.88879276 +0000 UTC]     return _unicodify_header_value(self.environ['HTTP_' + key])
[2018-12-03 06:56:21.888797545 +0000 UTC] KeyError: 'HTTP_X_MY_HEADER'
```